utils/s3/creds.hh

```diff
@@ -25,9 +25,13 @@ struct endpoint_config {
         // the security token, only for session credentials
         std::string session_token;
         std::string region;
+
+        std::strong_ordering operator<=> (const aws_config& o) const = default;
```

Comment:
`updateable_value_source::set(T)` compares the old value with the new one, presumably, the operator of `==` would suffice. so 
```c++
bool operator== (const aws_config& o) const = default;
```
would serve the purpose, right?

Reply:
Correct, equality operator would work too. Do you have any concerns about spaceship one in this place?

Reply:
If some of us uses regex when `==` is good enough, I complain. if some of us uses template when a plain function is good enough, I complain. If some of us uses shuttle when a scooter is enough, I complain.